xmtp_mls/src/client.rs

```diff
@@ -814,14 +814,30 @@ where
     pub(crate) async fn get_key_packages_for_installation_ids(
         &self,
         installation_ids: Vec<Vec<u8>>,
-    ) -> Result<Vec<VerifiedKeyPackageV2>, ClientError> {
-        let key_package_results = self.api_client.fetch_key_packages(installation_ids).await?;
+    ) -> Result<
+        HashMap<Vec<u8>, Result<VerifiedKeyPackageV2, KeyPackageVerificationError>>,
+        ClientError,
+    > {
+        let key_package_results = self
+            .api_client
+            .fetch_key_packages(installation_ids.clone())
+            .await?;
 
         let crypto_provider = XmtpOpenMlsProvider::new_crypto();
-        Ok(key_package_results
-            .values()
-            .map(|bytes| VerifiedKeyPackageV2::from_bytes(&crypto_provider, bytes.as_slice()))
-            .collect::<Result<_, _>>()?)
+
+        let results: HashMap<Vec<u8>, Result<VerifiedKeyPackageV2, KeyPackageVerificationError>> =
+            installation_ids
+                .iter()
+                .zip(key_package_results.values())
+                .map(|(installation_id, bytes)| {
+                    let parsed_result =
+                        VerifiedKeyPackageV2::from_bytes(&crypto_provider, bytes.as_slice())
+                            .map_err(|e| e);
```

Comment:
doesn't seem to do anything

Reply:
left out from the forced error return, good catch!